tools/mage/teardown.go

```diff
@@ -197,6 +204,129 @@ func destroyCfnStacks(awsSession *session.Session) error {
 	return nil
 }
 
+// The onboard stack has global IAM roles, ask user if they want to delete since this could affect other regions
+func deleteOnboardStack(awsSession *session.Session, results chan deleteStackResult) {
+	logger.Infof("deleting CloudFormation stack: %s", onboardStack)
+	cfClient := cloudformation.New(awsSession)
+	onboardStackExists, err := stackExists(cfClient, onboardStack)
+	if err != nil {
+		logger.Fatalf("error checking stack %s: %v", onboardStack, err)
+	}
+	if onboardStackExists {
+		auditRoleExists, err := roleExists(iam.New(awsSession), auditRole)
+		if err != nil {
+			logger.Fatalf("error checking audit role name %s: %v", auditRole, err)
+		}
+		if auditRoleExists {
+			answer := promptUser("Do you want to delete CloudSecurity roles (this can affect deployments in other regions)? (yes|no) ",
+				nonemptyValidator)
+			if strings.ToLower(answer) == "yes" {
+				logger.Infof("deleting stack %s", onboardStack)
+				go deleteStack(cfClient, aws.String(onboardStack), results) // can be done in background
+			} else {
+				results <- deleteStackResult{stackName: onboardStack, err: errors.Errorf("skipping dropping stack %s", onboardStack)}
```

Comment:
Returning an error here will cause the whole teardown to fail immediately (the main routine will log the result, which uses `logger.Fatalf` for errors).

I would just log a warning and then return a nil error

```suggestion
				logger.Warnf("skipping stack %s", onboardStack)
				results <- deleteStackResult{stackName: onboardStack, err: nil}
```

Reply:
done